### ISSUE TITLE
Fix S2589 FP: Break Inside Loop

### ISSUE THREAD

**Issue by groogiam** [SonarSource/sonar-dotnet#8434]
### Description

Please provide a succinct description of your issue. Also please provide the rule ID (e.g. S1234)

### Repro steps

The following example results in a false positive.

```csharp
async Task Main()
{
	MyResult result = null;
	
	for (var i = 0; i < 300; i++)
	{
		await Task.Delay(10000);

		result = IsOpFinished();

		if (result.Completed != null)
		{
			break;
		}
	}

	if (result?.Completed == null) //s2580 flagged here when it should not be
	{
		throw new TimeoutException();
	}
}

// You can define other methods, fields, classes and namespaces here\
public class MyResult
{
	public DateTime? Completed { get; set; }
}

public static MyResult IsOpFinished()
{
        //actual method implementation would eventually return a completed status
	return new MyResult();
}
```

### Expected behavior

Should not flag s2589

### Actual behavior

Flags s2589

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* C#/VB.NET Plugins version Sonarlint 7.4.0.80741
* Visual Studio version 17.8.2
* MSBuild / dotnet version  8.0.100
* SonarScanner for .NET version (if used)
* Operating System

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for submitting this issue. Confirmed as False Positive.
